src/app/shared/wysiwyg/prosemirror.markdown.editor.ts

```diff
@@ -20,29 +20,23 @@ export class ProseMirrorImage {
 export class ProseMirrorMarkdownEditor {
 
   view: MenuBarEditorView;
-  savedState: EditorState;
 
   constructor(private el: ElementRef,
               private value: string,
               private images: ProseMirrorImage[],
               private setModel: Function,
               private promptForLink: Function) {
-    this.savedState = EditorState.create(this.stateConfig());
-    this.view = new MenuBarEditorView(el.nativeElement, this.viewProps(this.savedState));
+    let state = EditorState.create(this.stateConfig());
+    this.view = new MenuBarEditorView(el.nativeElement, this.viewProps(state));
   }
 
-  update(images: ProseMirrorImage[]) {
-    this.images = images;
-    let state = this.view.editor.state.reconfigure(this.stateConfig());
-    this.view.update(this.viewProps(state));
-  }
-
-  resetEditor() {
-    this.view.updateState(this.savedState);
-  }
-
-  setSavedState() {
-    this.savedState = this.view.editor.state;
+  update(value: string, images?: ProseMirrorImage[]) {
```

Comment:
signature change, getting images array too.
not comparing the existing images to this, always just overwriting when passed in?
probably all that does is rebuild the menu, will have to check...